vault/resource_aws_secret_backend.go

```diff
@@ -99,31 +99,34 @@ func awsSecretBackendResource() *schema.Resource {
 	}
 }
 
-func mountMigrationCustomizeDiff(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) error {
-	if !diff.HasChange("path") {
+func mountMigrationCustomizeDiff_FieldPath(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+	return mountMigrationHelper(ctx, diff, meta, consts.FieldPath)
+}
+
+func mountMigrationHelper(_ context.Context, diff *schema.ResourceDiff, meta interface{}, mountField string) error {
+	if !diff.HasChange(mountField) {
 		return nil
 	}
 
-	o, _ := diff.GetChange("path")
+	o, _ := diff.GetChange(mountField)
 	if o == "" {
 		return nil
 	}
 
 	// Mount Migration is only available for versions >= 1.10
-	client, e := provider.GetClient(diff, meta)
-	if e != nil {
-		return e
+	minVersion, err := version.NewVersion(consts.VaultVersion10)
+	if err != nil {
+		return err
 	}
-
-	remountEnabled, _, err := semver.GreaterThanOrEqual(ctx, client, consts.VaultVersion10)
+	remountEnabled, _, err := provider.GreaterThanOrEqual(meta, minVersion)
```

Comment:
Since we are making a behavioral change here, I wonder if it would make sense to provide a boolean field for the user to opt out of the automatic remount, in the case where they are relying on the previous `ForceNew` behavior?

Reply:
Interesting, I think it might make sense if we can determine a case that the user wouldn't want their mounts to be moved and instead would want them to be destroyed and recreated.

I think what might help inform our decision is the fact that Vault itself provides no such feature/boolean field for users to opt out of remounting, and mount migration is the default and only option available for >= v1.10.0 ([docs for mount migration](https://www.vaultproject.io/api-docs/system/remount#move-backend)). This leads me to believe that we might be okay without the boolean flag field

Reply:
The remount is an explicit action, here it is happening behind the scenes. One consideration is that a remount operation may not migrate everything related to the mount. I believe that is a known limitation, so having the extra escape hatch might make sense here. The default is to perform a graceful remount, rather than a destroy before create operation.

Reply:
Ah, that is a good point. I think implementing a boolean field wouldn't be too bad. Are you okay with me implementing a `mustAddMountMigrationSchema` util? I could add that field to all backends that enable mount migration

Reply:
sure, sounds good to me!

Reply:
Added a `mustAddMountMigrationSchema` util that adds a boolean to disable remounts. An explicit test to disable remounts has been demonstrated in the Consul Secret backend tests